src/api/resources/collect_record.py

```diff
@@ -238,3 +230,40 @@ def missing(self, request, *args, **kwargs):
             missing_ids = [row[0] for row in cursor.fetchall()]
 
         return Response(dict(missing_ids=missing_ids))
+
+    @action(
+        detail=False, methods=["POST"], permission_classes=[ProjectDataAdminPermission]
+    )
+    def ingest(self, request, project_pk, *args, **kwargs):
+        supported_content_types = (
+            "application/csv",
+            "application/x-csv",
+            "text/csv",
+            "text/comma-separated-values",
+            "text/x-comma-separated-values",
+            "text/tab-separated-values",
+        )
+
+        protocol = request.data.get("protocol")
+        uploaded_file = request.FILES.get("file")
+        profile = request.user.profile
+        dryrun = truthy(request.data.get("dryrun"))
+
+        if protocol is None:
+            return Response("Missing protocol", status=400)
+
+        if uploaded_file is None:
```

Comment:
Don't know how it would be possible, but should we similarly protect against missing `profile.id`?

Reply:
I didn't think the user would get this far if they didn't have a `profile.id`, Permission would stop them